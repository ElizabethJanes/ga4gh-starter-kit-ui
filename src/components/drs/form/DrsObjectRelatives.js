import React from 'react';
import { 
    Container,
    Typography, 
    FormControl,
    TextField, 
    Grid, 
    FormGroup, 
    Button, 
    Divider, 
    Box, 
    RadioGroup, 
    Radio, 
    FormControlLabel, 
    IconButton, 
    MenuItem, 
    Tooltip,
    InputAdornment
} from '@material-ui/core';
import SpaceDivider from '../../common/SpaceDivider';
import RemoveItemButton from '../../common/RemoveItemButton';
import VerifyIdButton from './VerifyIdButton';
import AddItemButton from '../../common/AddItemButton';

const DrsObjectRelatives = props => {
    return (
        <FormGroup>
            <SpaceDivider />
            <Typography align='left' variant='h6'>{props.header}</Typography>
            {props.sectionDescription}
            {props.relatives.map((relative, index) => {
                return (
                    <FormGroup key={`props.relationship-${index}`} row>
                        <Grid container alignItems='center' spacing={4}>
                            <Grid item xs> 
                                <FormControl fullWidth>
                                    <TextField
                                        fullWidth
                                        variant='outlined'
                                        id={`ID_${props.relationship}${index}`} 
                                        label='Id' name={relative.id} value={relative.id} margin='normal' type='text' 
                                        onChange={e => {
                                            props.setRelativeId(index, e.target.value);
                                            props.unsetRelativeValidity(index);
                                        }}
                                        InputProps={
                                            {
                                                readOnly: props.readOnly,
                                                endAdornment: 
                                                    <InputAdornment position='end'>
                                                        <VerifyIdButton 
                                                            activeDrsObject={props.activeDrsObject}
                                                            relative={relative}
                                                            retrieveDrsObject={props.retrieveDrsObject}
                                                            readOnly={props.readOnly}
                                                            //drsObjectFunctions={props.drsObjectFunctions} 
                                                            //index={index} 
                                                            //property={props.relationship}
                                                        />
                                                    </InputAdornment>
                                            }
                                        }
                                    />
                                </FormControl>
                            </Grid>
                            <Grid item xs>
                                <FormControl fullWidth>
                                    <TextField variant='outlined' fullWidth id={`Name_${props.relationship}${index}`} 
                                    label='Name' margin='normal' name={relative.name} type='text' 
                                    value={relative.name} InputProps={{readOnly: true}}/>                            
                                </FormControl>
                            </Grid>

                            <RemoveItemButton
                                display={!props.readOnly}
                                objectName={props.objectName}
                                handleClick={() => props.removeRelative(index)}
                            />
                        </Grid>
                    </FormGroup>
                )
            })}
            <AddItemButton
                objectName={props.objectName}
                display={!props.readOnly}
                disabled={props.readOnly}
                handleClick={props.addRelative}
            />
        </FormGroup>
    )
}

export default DrsObjectRelatives;
