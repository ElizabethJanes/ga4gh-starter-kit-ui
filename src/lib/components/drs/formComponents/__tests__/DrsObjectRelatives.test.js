import React from 'react';
import renderer from 'react-test-renderer';
import DrsObjectRelatives from '../DrsObjectRelatives';

let zeroRelatives = [];
let multipleRelatives = [
    {id: 'ID123', name: 'name123'},
    {id: 'ID456', name: 'name456'},
    {id: 'ID789', name: 'name789'}
]
let children = {
    relationship: 'drs_object_children',
    objectName: 'child',
    header: 'Bundle Children'
}
let parents = {
    relationship: 'drs_object_parents',
    objectName: 'parent', 
    header: 'Bundle Parents'
}
let description='This is a description for snapshot testing.'
let showForm = {readOnly: true, formViewType: 'SHOW'}
let newForm = {readOnly: false, formViewType: 'NEW'}
let editForm = {readOnly: false, formViewType: 'EDIT'}

test('SHOW <DrsObjectRelatives/> should handle zero children', () => {
    const drsObjectRelatives = renderer
    .create(<DrsObjectRelatives 
        {...showForm}
        sectionDescription={description}
        relatives={zeroRelatives}
        {...children} />)
    .toJSON();
    expect(drsObjectRelatives).toMatchSnapshot();
})

/* test('SHOW <DrsObjectRelatives/> should handle multiple children', () => {
    const drsObjectRelatives = renderer
    .create(<DrsObjectRelatives 
        {...showForm}
        sectionDescription={description}
        relatives={multipleRelatives}
        {...children} />)
    .toJSON();
    expect(drsObjectRelatives).toMatchSnapshot();
}) */

test('NEW <DrsObjectRelatives/> should handle zero children', () => {
    const drsObjectRelatives = renderer
    .create(<DrsObjectRelatives 
        {...newForm}
        sectionDescription={description}
        relatives={zeroRelatives}
        {...children} />)
    .toJSON();
    expect(drsObjectRelatives).toMatchSnapshot();
})

/* test('NEW <DrsObjectRelatives/> should handle multiple children', () => {
    const drsObjectRelatives = renderer
    .create(<DrsObjectRelatives 
        {...newForm}
        sectionDescription={description}
        relatives={multipleRelatives}
        {...children} />)
    .toJSON();
    expect(drsObjectRelatives).toMatchSnapshot();
}) */

test('EDIT <DrsObjectRelatives/> should handle zero children', () => {
    const drsObjectRelatives = renderer
    .create(<DrsObjectRelatives 
        {...editForm}
        sectionDescription={description}
        relatives={zeroRelatives}
        {...children} />)
    .toJSON();
    expect(drsObjectRelatives).toMatchSnapshot();
})

/* test('EDIT <DrsObjectRelatives/> should handle multiple children', () => {
    const drsObjectRelatives = renderer
    .create(<DrsObjectRelatives 
        {...editForm}
        sectionDescription={description}
        relatives={multipleRelatives}
        {...children} />)
    .toJSON();
    expect(drsObjectRelatives).toMatchSnapshot();
}) */