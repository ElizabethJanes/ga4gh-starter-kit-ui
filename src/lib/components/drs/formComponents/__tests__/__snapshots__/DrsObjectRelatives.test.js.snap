// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EDIT <DrsObjectRelatives/> should handle zero children 1`] = `
<div
  className="MuiFormGroup-root"
>
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    <hr
      className="MuiDivider-root"
    />
  </div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
  >
    Bundle Children
  </h6>
  This is a description for snapshot testing.
  <div>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Add child"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`NEW <DrsObjectRelatives/> should handle zero children 1`] = `
<div
  className="MuiFormGroup-root"
>
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <hr
      className="MuiDivider-root"
    />
  </div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
  >
    Bundle Children
  </h6>
  This is a description for snapshot testing.
  <div>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Add child"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`SHOW <DrsObjectRelatives/> should handle zero children 1`] = `
<div
  className="MuiFormGroup-root"
>
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <hr
      className="MuiDivider-root"
    />
  </div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
  >
    Bundle Children
  </h6>
  This is a description for snapshot testing.
  <div />
</div>
`;
